bento_beacon/utils/katsu_utils.py

```diff
@@ -141,21 +141,30 @@ def get_katsu_config_search_fields():
 }
 
 
-# assume json query already validated
 # convert an individual beacon filter into bento format
 def bento_query_expression(q):
     # break up phenopackets property name with "#resolve" appended at the front
     katsu_key = ["#resolve", *q["id"].split(".")]
 
-    # extra handling for negation, "!" is always negated equality
+    beacon_value = q["value"]
+    has_wildcard = "%" in beacon_value
+
+    # reject meaningless cases
+    if has_wildcard and q["operator"] in ["<", "<=", ">", ">="]:
+        raise InvalidQuery("cannot interpret wildcard character '%' with an inequality operator (<, <=, >, >=)")
+
+    # separate handling for negation
     if q["operator"] == "!":
-        return ["#not", ["#eq", katsu_key, q["value"]]]
+        op = "#ilike" if has_wildcard else "#eq"
```

Comment:
should this be case sensitive or not? right now it's case sensitive with equals but insensitive otherwise

Reply:
Case-insensitive with wildcards is my best-guess reading of the documentation, since the spec itself says nothing. We can revisit this if it seems perverse.

Reply:
sounds good